### PR TITLE
Code commenting on the Python wrapper

### PR DESCRIPTION
**Pull request by Bash-04**
Created all code comments in the Python Wrapper + fixed hsv_to_rgb conversion function

### CODE REVIEW

**Comment by Bash-04** [comment]
@WouterPennings, I found out that the hsv_to_rgb conversion function wasn't given the expected results. So I recreated the code and commented out the old code. If I didn't do it right you can just change it back, if I did do it right then it's fixed now.

File: lib/wrappers/python/contourwall.py [hunk 1]
```diff
@@ -95,11 +145,16 @@ def show(self, sleep_ms:int=0, optimize:bool=True) -> None:
 
     def fill_solid(self, r: int, g: int, b: int) -> None:
         """
-        fill_solid is a function that fills the entire ContourWall with a single color. Each seperate LED will have the same color.
+        fill_solid is a function that fills the entire ContourWall with one single color.
```
Comment (WouterPennings) [inline lib/wrappers/python/contourwall.py:148]: This one can be yeeted, as it is the same as the one below it.

File: lib/wrappers/python/contourwall.py [hunk 2]
```diff
@@ -111,34 +166,99 @@ def drop(self) -> None:
         self._drop(ctypes.byref(self._cw_core))
 
 def hsv_to_rgb(hue: int, saturation: float, value: float) -> tuple[int, int, int]:
```
Comment (WouterPennings) [inline lib/wrappers/python/contourwall.py:168]: This function shoul be replaced with this. Make sure you improve the naming and such


```python
def hsv_to_rgb(h, s, v):
    h /= 360
    s /= 100
    v /= 100
    
    if s == 0.0: return v, v, v
        
    i = int(h*6.0) # XXX assume int() truncates!
    f = (h*6.0) - i
    i = i%6
    
    p = round((v*(1.0 - s)) * 255)
    q = round((v*(1.0 - s*f)) * 255)
    t = round((v*(1.0 - s*(1.0-f))) * 255)
    v = round(v*255)
    
    if i == 0:
        return v, t, p
    if i == 1:
        return q, v, p
    if i == 2:
        return p, v, t
    if i == 3:
        return p, q, v
    if i == 4:
        return t, p, v
    if i == 5:
        return v, p, q
```

File: lib/wrappers/python/contourwall.py [hunk 3]
```diff
@@ -111,34 +166,99 @@ def drop(self) -> None:
...
+    # elif i == 1:
+    #     return int(q * 255), int(value * 255), int(p * 255)
+    # elif i == 2:
+    #     return int(p * 255), int(value * 255), int(t * 255)
+    # elif i == 3:
+    #     return int(p * 255), int(q * 255), int(value * 255)
+    # elif i == 4:
+    #     return int(t * 255), int(p * 255), int(value * 255)
+    # else:
+    #     return int(value * 255), int(p * 255), int(q * 255)
```
Comment (WouterPennings) [inline lib/wrappers/python/contourwall.py:244]: Can be removed

File: lib/wrappers/python/demo.py [hunk 4]
```diff
@@ -2,6 +2,10 @@
 import time
 import sys
 
+def test_hsv_to_rgb():
+    # Convert HSV to RGB (the color RED)
+    print(hsv_to_rgb(0, 100, 100))
+
```
Comment (WouterPennings) [inline lib/wrappers/python/demo.py:8]: Should also be removed

**Review by WouterPennings — CHANGES_REQUESTED** [review]

File: lib/wrappers/python/contourwall.py (hunk 2 above)
Comment (WouterPennings) [reply lib/wrappers/python/contourwall.py:168]: make sure you also mention in a comment that we stole this implementation from Colorsys:

https://github.com/python/cpython/blob/3.12/Lib/colorsys.py

**Review by WouterPennings — APPROVED** [review]
Vere noice @Bash-04

**Pull request merged**